draft-ietf-oauth-device-flow.xml

```diff
@@ -415,33 +468,169 @@
+          that the device is in their possession.
+          </t>
+          <t>
+          The user code needs to have a long enough lifetime to be useable (allowing
+          the user to retrieve their secondary device, navigate to the verification URI, login, etc),
+          but should be sufficiently short to limit the
+          usability of a code obtained for phishing. This doesn't prevent a
+          phisher presenting a fresh token, particularly in the case they
+          are interacting with the user in real time, 
+          but it does limit the viability of codes sent over email or SMS.
+          </t> 
+        </section>
+        <section title="Non-confidential Clients" anchor="NonConfidentialClient">
+          <t>
+          Most device clients are incapable of being confidential clients, as secrets
+          that are statically included as part of an app distributed to
+          multiple users cannot be considered confidential.
+          For such clients, the recommendations of 
+          Section 5.3.1 of <xref target="RFC6819" /> and
+          Section 8.9 of <xref target="I-D.ietf-oauth-native-apps"/> apply.
+          </t>
+        </section>
+        <section title="Non-Visual Code Transmission" anchor="NonVisualCodeTransmission">
+          <t>
+          There is no requirement that the user code be displayed by the
+          device visually. Other methods of one-way communication can potentially be
+          used, such as text-to-speech audio, or Bluetooth LE. To mitigate an attack
+          where a malicious user can bootstrap their credentials on a device
+          not in their control, it is RECOMMENDED that any chosen communication 
+          channel only be accessible by people in close proximity. E.g., users
+          who can see, or hear the device, or within range of a short-range
+          wireless signal.
+          </t>
+        </section>
+
+    </section>
+
+    <section anchor="Usability" title="Usability Considerations">
+      <t>
+        This section is a non-normative discussion of usability considerations.
+      </t>
+      
+      <section anchor="UserCodeRecommendations" title="User Code Recommendations">
+        <t>
+          For many users, their nearest internet-connected device will be their
+          mobile phone, and typically these devices offer input methods that are
+          more time consuming than a computer keyboard to change the case or
+          input numbers. To improve usability (improving entry speed, and
+          reducing retries), these limitations should be taken into account
+          when selecting the user-code character set.
+        </t>
+        <t>
+          One way to improve input speed is to restrict the character set to
+          case-insensitive A-Z characters, with no digits. These characters can
+          typically be input on a mobile keyboard without using modifier keys.
```

Comment:
s/ typically be input on a mobile keyboard/ typically be entered on a mobile keyboard